### DISCUSSION TITLE
Optional braces: Is that intended?

### DISCUSSION THREAD

**Discussion by scf37** [scala/scala3#19963] [Feature Requests]
I find it weird I can write function arguments after `:` but can't write `case`
```scala
// ok
Map("hello" -> "world").map { case (key, value) => 
  key + value
}

// ok
Map("hello" -> "world").map : 
  case (key, value) => 
    key + value

// ok
Map("hello" -> "world").map : kv => 
  kv._1 + kv._2

// identifier expected but case found
Map("hello" -> "world").map : case (key, value) => 
  key + value

```

**Comment by SethTisue** [comment]
for tuples in particular, `case` isn't needed:

```
Welcome to Scala 3.4.0 (17.0.10, Java OpenJDK 64-Bit Server VM).

scala> Map("hello" -> "world").map: (key, value) => 
     |   key + value
     | 
val res0: scala.collection.immutable.Iterable[String] = List(helloworld)
```

as per https://docs.scala-lang.org/scala3/reference/other-new-features/parameter-untupling.html

yet one might still wonder about types other than tuples. I don't see anything about it in the long design discussion at https://contributors.scala-lang.org/t/make-fewerbraces-available-outside-snapshot-releases/5024/196

my best guess would be that it's considered satisfactory to put `case` on the next line, so e.g.

```
scala> case class C(x: Int)
// defined case class C
                                                                                                    
scala> List(C(0), C(1)).collect:
     |   case C(n) => n
     | 
val res4: List[Int] = List(0, 1)
```

this way of writing it has the advantage that you can add more `case`s, whereas if `case` were handled specially in `fewerBraces` syntax, it would only work for the single-`case` case

**Comment by EugeneFlesselle** [comment]
I don't particularly like this either, but yes it is a _known_ restriction.

**Comment by som-snytt** [comment]
Whenever I get a chance to use one-line catch
```
scala> try ??? catch case e: NotImplementedError => println("caught you")
caught you
```
I want to use one-line case everywhere.

```
scala> (42: Int) match case 42 =>
-- [E040] Syntax Error: ------------------------------------------------------------------------------------------------
1 |(42: Int) match case 42 =>
  |                ^^^^
  |                '{' expected, but 'case' found
```

`case` is the hardest of keywords, one may call it "case-hardened". I'd expect to use it in various one-liner idioms, although I haven't worked that out on a napkin.

https://en.wikipedia.org/wiki/Case-hardening
https://www.merriam-webster.com/dictionary/case-hardened

The dictionary says it can be taken metaphorically for "callous" or let's say "hard-boiled".

The other vocab word from the wiki article is "bloomery". I was unfamiliar with the metallurgy.
